go/types/list_iterator.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, verlion 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import (
+	"github.com/attic-labs/noms/go/d"
+)
+
+// ListIterator can be used to efficiently iterate through a Noms List starting at an index of your choice.
+type ListIterator struct {
+	*sequenceCursor
```

Comment:
yeah. I tried doing it as a type alias so I didn't have to have the extra struct, but there was a lot of `(*sequenceCursor)(li)` going on and that was ugly.